Fix rename/move on IFS directories

SanjulaGanepola opened the pull request:
### Changes

Fixes #2103

### How to test this PR

Examples:

1. Run the test cases
2. Expand view A and right click on the node
3. Run 'Execute Thing' from the command palette

### Checklist

* [x] have tested my change

sebjulliand commented on src/views/ifsBrowser.ts:
```diff
@@ -616,16 +616,19 @@ export function initializeIFSBrowser(context: vscode.ExtensionContext) {
-        vscode.window.showErrorMessage(t("ifsBrowser.moveIFS.errorMessage", t(String(node.contextValue)), t("file.unsaved.changes")));
-        return;
+      const oldFileTabs: vscode.Tab[] = [];
+      if (node.contextValue?.startsWith(`streamfile`)) {
```
It may be safer to check `file.type` here; if the `contextValue` changes in the future, this won't break.
```suggestion
      if (node.file.type === "streamfile") {
```

sebjulliand reviewed (changes requested):
Good catch @SanjulaGanepola!
I just have one suggestion. Otherwise it looks good!

sebjulliand reviewed (changes requested):
Code looks good and it works fine, thank you!
Now, to keep a good UX, could you add a check similar to the one done when renaming a file, to ensure every tab opened is saved before it gets renamed? When a folder gets renamed, checking if every tab opened is not dirty when its document path starts with `${node.file.path}/` (adding the / at the end to avoid false positives) should do the trick.
What do you think?

SanjulaGanepola commented:
Good idea to keep the behavior consistent between stream files and directories. Updated the PR.

chrjorgensen commented:
@sebjulliand Danish translation is in...

chrjorgensen reviewed (changes requested):
@SanjulaGanepola I found an (old?) error - to reproduce:
1. Open a file in a directory (do not change it)
2. Rename the directory containing the file
3. The new directory is opened as a file, causing an error message:

![billede](https://github.com/codefori/vscode-ibmi/assets/13275072/59391186-b65e-4b39-9614-61b4e502c7f8)

chrjorgensen commented:
@NicolasSchindler You changed the Danish translation file instead of the German translation file in commit [3fd6f90](https://github.com/codefori/vscode-ibmi/pull/2104/commits/3fd6f903f0b047f8ec1503a7117f805260e87f70) - could you please fix this? 😉

sebjulliand commented:
> @NicolasSchindler You changed the Danish translation file instead of the German translation file in commit [3fd6f90](https://github.com/codefori/vscode-ibmi/pull/2104/commits/3fd6f903f0b047f8ec1503a7117f805260e87f70) - could you please fix this? 😉

This one's on me since @NicolasSchindler couldn't commit directly on this branch? I'll fix it!

SanjulaGanepola commented:
> 3. The new directory is opened as a file, causing an error message:

@chrjorgensen Thanks for catching that issue. Fixed it -> [2ec6e40](https://github.com/codefori/vscode-ibmi/pull/2104/commits/2ec6e40514c47ab8d50651ce798fd80aa1198268)

FYI, also found another issue where the `mv` command would fail but no error was shown if for example you tried to move a file to a directory that does not exist. Fixed this as well -> [19552a5](https://github.com/codefori/vscode-ibmi/pull/2104/commits/19552a5714b7d0bcf0a6ad5c43a3579faea608ef)

worksofliam commented:
@SanjulaGanepola Please update the 'How to test' in the PR.

worksofliam commented:
@SanjulaGanepola 'Move/Rename...' works as expected. Do let me know if there is more than needs testing.

chrjorgensen reviewed (approved):
@SanjulaGanepola Looking great, working as expected and fixes the errors. 👍

SanjulaGanepola commented:
> @SanjulaGanepola 'Move/Rename...' works as expected. Do let me know if there is more than needs testing.

Sorry for the late reply. I see this was already merged, but yes only "Move/Rename" needed to be tested on directories
